cli/daemon/run/testdata/echo_client/client.ts

```diff
@@ -207,16 +207,16 @@ export namespace echo {
         /**
          * Body
          */
-        Struct: Data<Data<string, string>, number>
+        Struct: Data<(Data<string, string> | null), number>
 
-        StructPtr: Data<number, number>
-        StructSlice: Data<string, string>[]
-        StructMap: { [key: string]: Data<string, number> }
-        StructMapPtr: { [key: string]: Data<string, string> }
+        StructPtr: (Data<number, number> | null)
+        StructSlice: (Data<string, string> | null)[]
+        StructMap: { [key: string]: (Data<string, number> | null) }
+        StructMapPtr: ({ [key: string]: (Data<string, string> | null) } | null)
```

Comment:
Not sure about all of these. They're technically correct but very unlikely to match the user's intention. Can we do better somehow?

Reply:
Looking through all these changes, I feel like we're more technically correct at the expense of pragmatism.

I think we should think through these changes and see if we can do a better job inferring user intent somehow.

One idea is to define an `Option[T]` type and ask users to use that to communicate that something is optional instead of `*T`.

Another idea is to default to assuming pointers are non-null in certain (or all?) cases and add a validation struct tag that communicates that it's nullable/optional.

Reply:
The user could _not_ use a pointer then...

Reply:
When we add the validate system, a `validate:"present"` flag could be used to signal intention here.

But the biggest usecase I see for pointers on API parameters and response types (in the real world) is this:

```go
type Params {
    Index *int `json:",omitempty"`
}
```

This allows the API to understand the difference between `0` and `undefined` / `null`

Reply:
Exactly, so what can we do to more  accurately infer the user's intent?

Reply:
For now we've decided to defer this so we can think through a better design. (ENC-827).